doc/api/url.md

```diff
@@ -769,35 +869,75 @@ Returns `true` if there is at least one name-value pair whose name is `name`.
 
 Returns an ES6 Iterator over the names of each name-value pair.
 
+```js
+const { URLSearchParams } = require('url');
+const params = new URLSearchParams('foo=bar&foo=baz');
+for (const name of params.keys()) {
+  console.log(name);
+}
+  // Prints:
+  // foo
+  // foo
+```
+
 #### urlSearchParams.set(name, value)
 
 * `name` {String}
 * `value` {String}
 
-Remove any existing name-value pairs whose name is `name` and append a new
-name-value pair.
+Sets the value in the `URLSearchParams` object associated with `name` to
+`value`. If there are any pre-existing name-value pairs whose names are `name`,
+set the first such pair's value to `value` and remove all others. If not,
+append the name-value pair to the query string.
+
+```js
+const { URLSearchParams } = require('url');
+
+const params = new URLSearchParams();
+params.append('foo', 'bar');
+params.append('foo', 'baz');
+params.append('abc', 'def');
+console.log(params.toString());
+  // Prints foo=bar&foo=baz&abc=def
+
+params.set('foo', 'def');
+params.set('xyz', 'opq');
+console.log(params.toString());
+  // Prints foo=def&abc=def&xyz=opq
+```
 
 #### urlSearchParams.toString()
 
 * Returns: {String}
 
-Returns the search parameters serialized as a URL-encoded string.
+Returns the search parameters serialized as a percent-encoded string.
```

Comment:
Probably "as percent-encoded strings" or "as a string of percent-encoded characters", or even "percent-encoded bytes"? "a percent-encoded string" sounds a bit weird given that they would be not just be encoded but also joined with `&` and `=`s.

Another way to explain it would be to mention that it is somewhat similar to `querystring.stringify()`.